sync/src/relayer/compact_block_process.rs

```diff
@@ -197,40 +210,44 @@ impl<'a, CS: ChainStore + 'static> CompactBlockProcess<'a, CS> {
                 }
             }
         }
-        if !missing_indexes.is_empty() {
-            if !self
-                .relayer
-                .shared()
-                .write_inflight_blocks()
-                .insert(self.peer, block_hash.to_owned())
-            {
-                debug_target!(
-                    crate::LOG_TARGET_RELAY,
-                    "BlockInFlight reach limit or had requested, peer: {}, block: {:x}",
-                    self.peer,
-                    block_hash,
-                );
-                return Ok(());
-            }
 
-            let fbb = &mut FlatBufferBuilder::new();
-            let message = RelayMessage::build_get_block_transactions(
-                fbb,
-                &block_hash,
-                &missing_indexes
-                    .into_iter()
-                    .map(|i| i as u32)
-                    .collect::<Vec<_>>(),
+        // The missing_indexes may never be empty
+        if missing_indexes.is_empty() {
+            return Err(Error::Internal(Internal::MissingIndexesEmpty).into());
+        }
```

Comment:
`if missing_indexes.is_empty()` means `Ok(())`, not an error.

Reply:
The `missing_indexes` should never be empty. Otherwise, there would be some logical error in `reconstruct_block`.